addons/stock/models/stock_picking.py

```diff
@@ -204,6 +204,12 @@ def _onchange_show_operations(self):
         if self.show_operations and self.code != 'incoming':
             self.show_reserved = True
 
+    @api.constrains('default_location_src_id', 'default_location_dest_id')
+    def _check_default_location(self):
+        for record in self:
+            if record.code == 'mrp_operation' and record.default_location_dest_id.scrap_location:
+                raise ValidationError(_("You cannot set a location of type Scrap, to a manufacturing type operation."))
```

Comment:
Sorry, the English is this will also need to be updated to make sense. `You cannot set a scrap location as the destination location for a manufacturing type operation.`